src/containers/org/members/OrgMemberContainer.js

```diff
@@ -85,53 +92,83 @@ const OrgMemberContainer = ({
+                      member={member}
+                      organizationId={organizationId}
+                      roles={organization?.roles} />
+                </StackGrid>
+                <StackGrid>
+                  <Typography variant="h2">Data Sets</Typography>
+                  <Typography>Click on a dataset to manage permissions.</Typography>
+                </StackGrid>
+                <PermissionsActionsGrid
+                    assignPermissionsText="Add dataset"
+                    onChangeFilterByPermissionTypes={setFilterByPermissionTypes}
+                    onChangeFilterByQuery={setFilterByQuery}
+                    onClickAssignPermissions={() => setIsVisibleAddDataSetModal(true)} />
                 <DataSetPermissionsContainer
+                    filterByPermissionTypes={filterByPermissionTypes}
+                    filterByQuery={filterByQuery}
                     organizationId={organizationId}
                     onSelect={setSelection}
                     principal={memberPrincipal}
                     selection={selection} />
               </StackGrid>
-            </StackGrid>
-          </AppContentWrapper>
-        </ContentColumn>
-        {
-          selection && (
-            <PanelColumn>
-              <PermissionsPanel
-                  dataSetId={selection.dataSetId}
-                  key={`${selection.dataSetId}-${selection.permissionType}`}
-                  onClose={handleOnClosePermissionsPanel}
-                  principal={memberPrincipal}
-                  permissionType={selection.permissionType} />
-            </PanelColumn>
-          )
-        }
-      </AppContentGrid>
+            </AppContentWrapper>
+          </ContentColumn>
+          {
+            selection && (
+              <PanelColumn>
+                <PermissionsPanel
+                    dataSetId={selection.dataSetId}
+                    key={`${selection.dataSetId}-${selection.permissionType}`}
+                    onClose={handleOnClosePermissionsPanel}
+                    principal={memberPrincipal}
+                    permissionType={selection.permissionType} />
+              </PanelColumn>
+            )
+          }
+        </AppContentGrid>
+        <Modal
+            isVisible={isVisibleAddDataSetModal}
+            onClose={() => {}}
+            shouldCloseOnOutsideClick={false}
+            viewportScrolling
+            withFooter={false}
+            withHeader={false}>
+          <StepsController>
+            {
+              ({ step, stepBack, stepNext }) => (
+                <AssignPermissionsToDataSet
+                    onClose={() => setIsVisibleAddDataSetModal(false)}
+                    organizationId={organizationId}
+                    principal={memberPrincipal}
+                    step={step}
+                    stepBack={stepBack}
+                    stepNext={stepNext} />
+              )
+            }
+          </StepsController>
```

Comment:
StepsController should probably use a render prop instead of intercepting `children` as it is unclear which scope step, stepBack, stepNext come from. also would this still work if there are multiple children? if not, we should try not to break typically sanctioned jsx patterns.

also it seems AssignPermissionsToDataSet is not designed to be rendered without the steps. Consider moving the StepsController to be contained inside the component.

Reply:
I feel like we're going in circles here. Let's talk.